add script to lookup account by email address

andrewpearce-digital opened the pull request:
# Purpose

PM and UR want to be able to help with customer queries and look up account details so they can do so.

Fixes UML-1188

## Approach

Scan, filter and get items.

## Checklist

* [x] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] I have added welsh translation tags and updated translation files
* [ ] The product team have tested these changes

GemTay commented on scripts/analysis_scripts/lookup_account.py:
```diff
@@ -0,0 +1,161 @@
+        )
+        self.aws_iam_session = session
+
+    def get_viewer_codes(self):
```
this function needs renaming

andrewpearce-digital replied:
Fixed method and class naming

GemTay commented on scripts/analysis_scripts/lookup_account.py:
```diff
@@ -0,0 +1,161 @@
+    def write_csv(self,email_address):
+        print('Collecting data and writing CSV...')
+        n = 0
+        with open('some.csv', 'w', newline='') as f:
```
the csv file could do with a better name

andrewpearce-digital replied:
fixed CSV file naming now to be descriptive and include the date it was generated.

GemTay reviewed (changes requested):
just some renaming needed
Also, did you test on a case where the LastLogin attribute doesnt exist?

andrewpearce-digital commented:
> just some renaming needed
> Also, did you test on a case where the LastLogin attribute doesnt exist?

Thanks for the reminder. Added that in now. Will default to Never logged in if there's no last login date, and I've removed the filtering by login date as it's not necessary now that the output is updated.

GemTay commented on scripts/analysis_scripts/lookup_account.py:
```diff
@@ -0,0 +1,133 @@
+            writer = csv.writer(
+                csv_file, quoting=csv.QUOTE_NONNUMERIC)
+            writer.writerow(["email", "last_login_datetime","lpas_added"])
+            viewer_codes = self.get_actor_users()
```
rename this variable too, I missed that before, sorry!

GemTay reviewed (changes requested):
spotted one more thing that needs renaming (sorry!), everything else looks great :)

GemTay reviewed (approved):
nice job!
